OreNoTask/app/models/task.rb

```diff
@@ -1,4 +1,16 @@
 # frozen_string_literal: true
 
 class Task < ApplicationRecord
+  def start_end_check?
+    return if self.start_at.blank? || self.due_date_at.blank?
+
+    errors.add(:due_date_at, I18n.t('dictionary.messages.invalid_date_diff')) unless
+      self.start_at < self.due_date_at
+  end
+
+  validates :name, { presence: true, length: { maximum: 50 } }
+  validates :description, length: { maximum: 2000 }
+  validates :start_at, presence: true, date: true
+  validates :due_date_at, presence: true, date: true
+  validate :start_end_check?
```

Comment:
バリデーションの細かなロジックよりも各項目へのバリデーションの概要が先にきていた方がいいかなと思うので、順番を以下のように入れ替えていただけると嬉しいです
特にルールがあるわけではないですが、そのように書かれているコードが多いかなと思っている程度になりますmm
```suggestion
  validates :name, { presence: true, length: { maximum: 50 } }
  validates :description, length: { maximum: 2000 }
  validates :start_at, presence: true, date: true
  validates :due_date_at, presence: true, date: true
  validate :start_end_check?
  
  def start_end_check?
    return if self.start_at.blank? || self.due_date_at.blank?

    errors.add(:due_date_at, I18n.t('dictionary.messages.invalid_date_diff')) unless
      self.start_at < self.due_date_at
  end
```

Reply:
修正しました！